### ISSUE TITLE
[Bug🐛] Distinguish late client responses and immediate-shutdown work from transport faults

### ISSUE THREAD

**Issue by mxsm** [mxsm/rocketmq-rust#10621]
### Prerequisites

- [x] I have searched existing issues and discussions
- [x] This is a bug, not a question (use Discussions for questions)

### Environment

- OS: Windows 11 during reproduction; the defect is platform independent
- RocketMQ Rust: main after #10620
- Rust: workspace toolchain

### Bug Description

Two warnings report ordinary shutdown behaviour as a fault, so an operator cannot tell them apart from real problems.

The first comes from `ClientRoute::response` in `rocketmq-transport/src/clients/client.rs`. It calls the pending-request table and warns when the call reports no match:

```text
WARN rocketmq_transport::clients::client:269: received client response without a matching pending request code=0 session_id=2
```

A response legitimately arrives after its request already settled. The entry leaves the table through three expected paths: the caller stops waiting and drops `PendingRequestGuard`, which completes the request as `Cancelled`; the owning route retires the owner while draining, which completes every request for that owner as `SessionClosed`; or the request deadline elapses. Any client that stops sending and then shuts down produces this warning every time, because the requests it abandoned are still being answered by the broker.

The same boolean also covers a structurally different case: an owner that belongs to another pending-request table generation. That indicates a routing or generation mistake and is worth a warning. Because both cases return the same value, the actionable one is buried in expected noise. The server side has the same shape in `rocketmq-transport/src/dispatch/processor_adapter.rs`, which warns that an unmatched response frame was dropped.

The second comes from `RemotingClient::shutdown()` in `rocketmq-transport/src/clients/rocketmq_tokio_client/lifecycle.rs`, the compatibility wrapper that does not wait:

```text
WARN rocketmq_transport::clients::rocketmq_tokio_client::lifecycle:892: RemotingClient background task shutdown report is unhealthy
```

The wrapper calls `shutdown_now()`, which cancels and aborts without waiting, and then warns whenever the resulting `ShutdownReport` is not healthy. `record_task_outcomes` in `rocketmq-runtime/src/task_group.rs` counts every task still present in the registry at the moment the report is taken as `leaked` or `detached_still_running`. A shutdown that deliberately does not wait therefore reports exactly that condition whenever anything is still winding down, and the warning names no counter, so it reads as a fault when it is a property of the call.

### Steps to Reproduce

1. Start a local NameServer and Broker.
2. Run the quickstart client producer (`rocketmq-client/examples/quickstart/producer.rs`) against it.
3. Stop it while sends are in flight, so the client shuts down with requests the broker is still answering.
4. Observe the first warning as the responses arrive during shutdown, and the second at the end of the client shutdown sequence.

Both were reproduced this way against a local two-broker cluster, from `producer.shutdown()` through `example_runtime.shutdown()`.

### Expected vs Actual

**Expected:** A response that arrives after its request settled is reported at debug level, because the caller already has its answer, and a response for a foreign pending-request owner still warns. The same distinction applies to the server-side frame path. An immediate, non-waiting shutdown reports faults, meaning explicit failures and panics aggregated over the group tree, and names the counters it observed; work that is merely still winding down is not a fault for a call that does not wait. Graceful shutdown reports keep their current meaning, because they observe tasks after waiting for them.

**Actual:** Both an expected late response and a structural owner mismatch warn identically, and every immediate shutdown that leaves work winding down warns that the report is unhealthy without saying why.

### Logs/Screenshots

The two messages above are the whole defect. Both were captured from the reproduction, together with the version of the same scenario after the fix, where neither appears.

### Additional Context

The change covers the transport client and server response paths plus the immediate-shutdown wrapper in `rocketmq-transport`. The pending-request table gains a small result type so the two cases stop sharing one boolean; the runtime report keeps its current fields and meanings, and the fault decision is local to the caller that knows whether it waited.

Out of scope: changing `ShutdownReport`, its counters, or `is_healthy()`, which other callers rely on; and `ShutdownReport::log_if_unhealthy()`, which logs a whole report as JSON including task names, and deserves its own review rather than being folded in here.

Validation expected for the change:

```bash
cargo fmt -p rocketmq-transport -- --check
cargo clippy -p rocketmq-transport --no-deps --all-targets --features test-support -- -D warnings
cargo test -p rocketmq-transport --lib
```

---
